externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala

```diff
@@ -100,4 +108,26 @@ class ExecuteStatement(
       executeStatement()
     }
   }
+
+  private def addTimeoutMonitor(): Unit = {
+    if (queryTimeout > 0) {
+      val timeoutExecutor = Executors.newSingleThreadScheduledExecutor()
```

Comment:
Use `NamedThreadFactory` to create thread

Reply:
org.apache.kyuubi.util.ThreadUtils$#newDaemonSingleThreadScheduledExecutor